### PR TITLE
Arsenal - Exit refresh early if no display is found

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Title.
- Wait a frame before display no virtual items notification when removing a box, in case the removal is being done during display close.

@Mike-MF please test if this handles your case

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/arsenal/XEH_postInit.sqf [hunk 1]
```diff
@@ -22,7 +22,7 @@ GVAR(lastSortDirectionRight) = DESCENDING;
     params ["_object"];
 
     // If the arsenal is already open, refresh arsenal display
-    if (!isNil QGVAR(currentBox) && {GVAR(currentBox) isEqualTo _object}) then {
+    if (!isNull GVAR(currentBox) && {GVAR(currentBox) isEqualTo _object}) then {
```
Comment (LinkIsGrim) [inline addons/arsenal/XEH_postInit.sqf:25]: `GVAR(currentBox)` will only be nil during preInit

File: addons/arsenal/functions/fnc_removeBox.sqf [hunk 2]
```diff
@@ -42,8 +42,13 @@ if (_global && {isMultiplayer} && {!isNil "_id"}) then {
...
 
 // If the arsenal is already open and not ignoring content (see FUNC(openBox)), close arsenal display
-if (!isNil QGVAR(currentBox) && {GVAR(currentBox) isEqualTo _object} && {isNil QGVAR(ignoredVirtualItems)}) then {
-    [LLSTRING(noVirtualItems), false, 5, 1] call EFUNC(common,displayText);
-    // Delay a frame in case this is running on display open
-    [{(findDisplay IDD_ace_arsenal) closeDisplay 0}] call CBA_fnc_execNextFrame;
+if (!isNull GVAR(currentBox) && {GVAR(currentBox) isEqualTo _object} && {isNil QGVAR(ignoredVirtualItems)}) then {
+    // Delay a frame in case this is running on display open/close
+    [{
+        private _display = findDisplay IDD_ace_ace_arsenal;
```
Comment (Mike-MF) [inline addons/arsenal/functions/fnc_removeBox.sqf:48]:
```suggestion
        private _display = findDisplay IDD_ace_arsenal;
```

File: addons/arsenal/functions/fnc_removeBox.sqf [hunk 3]
```diff
@@ -42,8 +42,13 @@ if (_global && {isMultiplayer} && {!isNil "_id"}) then {
 };
 
 // If the arsenal is already open and not ignoring content (see FUNC(openBox)), close arsenal display
-if (!isNil QGVAR(currentBox) && {GVAR(currentBox) isEqualTo _object} && {isNil QGVAR(ignoredVirtualItems)}) then {
-    [LLSTRING(noVirtualItems), false, 5, 1] call EFUNC(common,displayText);
-    // Delay a frame in case this is running on display open
-    [{(findDisplay IDD_ace_arsenal) closeDisplay 0}] call CBA_fnc_execNextFrame;
+if (!isNull GVAR(currentBox) && {GVAR(currentBox) isEqualTo _object} && {isNil QGVAR(ignoredVirtualItems)}) then {
```
Comment (johnb432) [inline addons/arsenal/functions/fnc_removeBox.sqf:45]: I think this can be called in preInit, so I would stick with the `!isNil` check.

**Review by johnb432 — APPROVED** [review]
LGTM

**Pull request merged**